dodiscover/toporder/utils.py

```diff
@@ -0,0 +1,119 @@
+from typing import List
+
+import networkx as nx
+import numpy as np
+import pandas as pd
+from numpy.typing import NDArray
+
+
+def full_dag(top_order: List[int]) -> NDArray:
+    """Find adjacency matrix of the fully connected DAG from the topological order.
+
+    Parameters
+    ----------
+    top_order : List[int]
+        Topological order of nodes in a causal graph.
+
+    Returns
+    -------
+    A : np.ndarray
+        The fully connected adjacency matrix encoding the input topological order.
+    """
+    d = len(top_order)
+    A = np.zeros((d, d))
+    for i, var in enumerate(top_order):
+        A[var, top_order[i + 1 :]] = 1
+    return A
+
+
+def full_adj_to_order(A: NDArray) -> List[int]:
+    """Find topological ordering from the adjacency matrix A.
+
+    Parameters
+    ----------
+    A : np.ndarray
+        The fully connected adjacency matrix encoding the input topological order.
+
+    Returns
+    -------
+    top_order : List[int]
+        Topological order encoding of the input fully connected adjacency matrix.
+    """
+    order = list(A.sum(axis=1).argsort())
+    order.reverse()
```

Comment:
why is reverse necessary? Can you add an inline comment?

Reply:
Reverse is necessary because the topological order is meant to have source first, leaves last (and thus I implemented and used it according to this convention). I will add a comment on that